packages/sourcecred/src/util/batch.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+/**
+Returns an array of arrays that contains all of the items in the original
+array parameter, but batched into arrays no larger than the batchSize.
+Example:
+  batch([1,2,3,4,5], 2) = [[1,2], [3,4], [5]]
+*/
+export function batchArray<T>(
+  array: $ReadOnlyArray<T>,
+  batchSize: number
+): $ReadOnlyArray<$ReadOnlyArray<T>> {
+  const result = [];
+  let backlog = array;
+  while (backlog.length) {
```

Comment:
nit: I've tried to come up with a functional way without worrying about length that satisfies most of corner cases, then found `reduce` function: it's not well-documented on flow.
```js
function batchArray<T>(array: $ReadOnlyArray<T>, batchSize: number): $ReadOnlyArray<$ReadOnlyArray<T>> {
  return array.reduce(
    (previousValue, currentItem, idx) =>
      (idx % batchSize === 0 // simply, should we start a new array item or not.
        ? previousValue.push([currentItem]) // yes? okay, wrap the current item into array.
        : previousValue[previousValue.length - 1].push(currentItem)) &&
      previousValue,
    <T>[] // initial value incorrect syntax
  );
}
``` 

consider it as an alternative, but _personally_ I think loops are source of evil.

Reply:
Clever, I think that reduces readability significantly. It's more understandable as written--a batching method working at the granularity of batches.